<!DOCTYPE html>
<html lang="en">
  <head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    
    <style>
      div {
        font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
         
      }

      pre, code {
        color: rgb(24, 174, 119);
      }

      #content {
        background-color: white;
        margin-left: 170pt;
        margin-right: 170pt;
        margin-top: 50pt;
      }
      #date {
        color: gray;
        font-size: small;
      }
    </style>
    <script type="text/javascript">
      window.MathJax = {
        tex: {
          inlineMath: [
            ["$", "$"],
            ["\\(", "\\)"],
          ], 
        },
      };
    </script>
    <script
      id="MathJax-script"
      async
      src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"
    ></script>
  </head>
  <body>
    <div id="content">
      <h1>Differential Gene Expression (DGE) Analysis in R</h1>
      
      <hr>
<h1 id="background-and-motivation">Background and motivation</h1>
<p>In an RNA-seq experiment, we may have samples from different &ldquo;treatments&rdquo; or &ldquo;groups&rdquo;. For example, we could have a total of $10$ samples, with $5$ samples coming from the &ldquo;diseased&rdquo; group and the other $5$ samples coming from the healthy group &ndash; &ldquo;diseased&rdquo; in this context could refer to any disease (e.g. <a href="https://www.cdc.gov/copd/index.html">COPD</a>). We may have more or less than two treatments. This is just a toy example.</p>
<p>With DGE analysis, we can <em>analyze</em> RNA-seq data to identify which genes are <em>differentially expressed</em> across the treatments &ndash; hence the name.</p>
<p>Suppose we have two samples $S_{1}$ and $S_{2}$ from different treatments, and for each we measure the expression level of gene $G_{1}$ and $G_{2}$, amongst othere genes. If the expression of $G_{1}$ is &ldquo;roughly&rdquo; the same in both $S_{1}$ and $S_{2}$, then we say that $G_{1}$ is not differentially expressed, at least with respect to these two samples. Suppose on the other hand that $G_{2}$ <em>is</em> differentially expressed &ndash; say it is more expressed in the sample $S_{1}$ from the diseased group than it is in the sample $S_{2}$ from the healthy group. This would indicate that $G_{2}$ somehow contributes to the development of the disease we are examining. This is useful knowledge because we can know refer back to biological literature about the function of gene $G_{2}$ in the organism we are studying and understand how that relates to the symptoms expressed by the disease.</p>
<h1 id="dge-analysis-in-r-with-edger-and-limma-voom">DGE Analysis in R with <code>edgeR</code> and <code>limma-voom</code></h1>
<p>In an RNA-seq experiment, DGE analysis is performed downstream of forming a count matrix that quantifies gene expression level (per gene, per sample)<sup id="fnref:1"><a href="#fn:1" class="footnote-ref" role="doc-noteref">1</a></sup>. <code>limma</code> is an R (Bioconductor) package for the analysis of microarray data. <code>voom</code> is a function in the <code>limma</code> package that transforms the count-data from an RNA-seq experiment into a specific object which can then be treated by <code>limma</code> functions as microarray data. This is stated in the <code>limma</code> <a href="https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf">user guide</a>:</p>
<blockquote>
<p><em>&ldquo;From version 3.9.19, limma includes functions to analyse RNA-seq experiments, demonstrated
in Case Study 11.8. The approach is to convert a table of sequence read counts into an expression
object which can then be analysed as for microarray data&rdquo;</em></p>
</blockquote>
<h2 id="log-normalization-of-rna-seq-count-data">Log-normalization of RNA-seq count data</h2>
<p><code>voom</code> returns an <code>EList</code> (i.e. Expression List) object with a component <code>E</code> which is the numeric matrix of the <em>log-normalized</em> counts <sup id="fnref:2"><a href="#fn:2" class="footnote-ref" role="doc-noteref">2</a></sup><sup id="fnref:3"><a href="#fn:3" class="footnote-ref" role="doc-noteref">3</a></sup>. This <em>log-normalization</em> is described in Law et al (2014) <sup id="fnref:4"><a href="#fn:4" class="footnote-ref" role="doc-noteref">4</a></sup> and explained below.</p>
<p>Supposing that we have a count matrix $C$ for $n$ samples $i=1$ to $i=n$ and $G$ genes $g=1$ to $g=G$. Let $r_{i,g}$ denote the number of reads aligned to gene $g$ for sample $i$. Define the <em>library size</em> of a sample $i$ as the sum of the counts aligned to all the genes:
$$
R_{i} = \sum_{g=1}^{G}r_{i,g}
$$</p>
<p>Now, given the count matrix $C$, <code>voom</code> applies a tranformation $\mathcal{L}$ to each read count $r_{i,g}$ to produce a <em>log-normalized</em> read count $y_{i,g}$ using the formula:
$$
y_{i,g} = \log_{2}\Big( \frac{r_{i,g} + 0.5}{R_{i} + 1.0} \times 10^{6}\Big)
$$</p>
<p>Thus, the matrix $E$ of log-normalized counts can be expressed as:
$$
E = \mathcal{L}(C)
$$</p>
<p>This transformation is only relevant when the library sizes between the samples differ significantly<sup id="fnref:5"><a href="#fn:5" class="footnote-ref" role="doc-noteref">5</a></sup></p>
<h2 id="data-visualization-for-dge-analysis">Data Visualization for DGE Analysis</h2>
<p>Once we have obtained a list of differentially expressed genes, we can plot the gene-expression levels (as log-cpm normalized values or otherwise) across the treatments, for each sample in each treatment. Doing so gives us an idea of</p>
<hr>
<p><code>voom</code> takes in as input un-normalized counts, which can possibly be estimated counts (i.e. not necessarily whole numbers), as generated by <code>Salmon</code> and <code>tximport</code>.</p>
<p>Naturally, one might ask what is so special about RNA-seq count data that it needs to undergo special processing (via <code>voom</code>) before being passed into <code>limma</code> functions. In other words, how is RNA-seq data different from microarray data? For one, RNA-seq data tends to follow the <a href="https://www.sciencedirect.com/topics/mathematics/negative-binomial-distribution">Negative Binomial Distribution</a> when the <code>limma</code> functions made for microarray data expect normally distributed data. The purpose of <code>limma</code> and <code>voom</code></p>
<p>The complicated part is <em>how</em> exactly <code>voom</code> transforms the RNA-seq count data so it becomes acceptable for <code>limma</code> functions.</p>
<div class="footnotes" role="doc-endnotes">
<hr>
<ol>
<li id="fn:1">
<p>Refer to my article on <a href="http://localhost:1313/posts/salmon-tximport/" title="Salmon and tximport"><code>Salmon</code> and <code>tximport</code></a> for an explanation of how to derive a count matrix from <code>Salmon</code>&rsquo;s transcript-level quantification using <code>tximport</code>&#160;<a href="#fnref:1" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
<li id="fn:2">
<p>From page 250 of the <a href="https://bioconductor.org/packages/release/bioc/manuals/limma/man/limma.pdf"><code>limma</code> reference manual</a>&#160;<a href="#fnref:2" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
<li id="fn:3">
<p>Note that in statistics, (numeric) data is often log-scaled when the data can take on large values&#160;<a href="#fnref:3" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
<li id="fn:4">
<p>In the &ldquo;Materials and methods&rdquo; section of their paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29">voom: precision weights unlock linear model analysis tools for RNA-seq read counts</a>&#160;<a href="#fnref:4" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
<li id="fn:5">
<p>From page 72 of the <a href="https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf"><code>limma</code> user guide</a>&#160;<a href="#fnref:5" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
</ol>
</div>

    </div>
  </body>
</html>
