+++
title = 'Differential Gene Expression (DGE) Analysis in R'
date = 2024-09-23T12:20:07-07:00
draft = true
+++

---

# Background
In an RNA-seq experiment, we may have samples from different "treatments" or "groups". For example, we could have a total of $10$ samples, with $5$ samples coming from the "diseased" group and the other $5$ samples coming from the healthy group -- "diseased" in this context could refer to any disease (e.g. [COPD]()). We may have more or less than two treatments. This is just a toy example.

With DGE analysis, we can identify which genes are *differentially expressed* across the treatments. 

Suppose we have two samples $S_{1}$ and $S_{2}$ from different treatments, and for each we measure the expression level of gene $G_{1}$ and $G_{2}$, amongst othere genes. If the expression of $G_{1}$ is "roughly" the same in both $S_{1}$ and $S_{2}$, then we say that $G_{1}$ is not differentially expressed, at least with respect to these two samples. Suppose on the other hand that $G_{2}$ *is* differentially expressed -- say it is more expressed in the sample $S_{1}$ from the diseased group than it is in the sample $S_{2}$ from the healthy group. This would indicate that $G_{2}$ somehow contributes to the development of the disease we are examining. This is useful knowledge because we can know refer back to biological literature about the function of gene $G_{2}$ in the organism we are studying and understand how that relates to the symptoms expressed by the disease.

# DGE Analysis in R with `edgeR` and `limma-voom`   
In an RNA-seq experiment, DGE analysis is performed downstream of forming a count matrix that quantifies gene expression level (per gene, per sample)[^1]. `limma` is an R (Bioconductor) package for the analysis of microarray data. `voom` is a function in the `limma` package that transforms the count-data from an RNA-seq experiment into a specific object which can then be interacted with by `limma` functions as microarray data. This is stated in the `limma` [user guide](https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf): 
> *"From version 3.9.19, limma includes functions to analyse RNA-seq experiments, demonstrated
in Case Study 11.8. The approach is to convert a table of sequence read counts into an expression
object which can then be analysed as for microarray data"*

Specifically, `voom` returns an `EList` (i.e. Expression List) object with a component `E` which is the numeric matrix of normalized expression values (on the log2 scale).[^2][^3]. `voom` takes in as input un-normalized counts, which can possibly be estimated counts (i.e. not necessarily whole numbers), as generated by `Salmon` and `tximport`.


Naturally, one might ask what is so special about RNA-seq count data that it needs to undergo special processing (via `voom`) before being passed into `limma` functions. In other words, how is RNA-seq data different from microarray data? For one, RNA-seq data tends to follow the [Negative Binomial Distribution](https://www.sciencedirect.com/topics/mathematics/negative-binomial-distribution) when the `limma` functions made for microarray data expect normally distributed data. The purpose of `limma` and `voom` 

The complicated part is *how* exactly `voom` transforms the RNA-seq count data so it becomes acceptable for `limma` functions.







[^1]: Refer to my article on [`Salmon` and `tximport`]({{<ref "posts/salmon-tximport">}} "Salmon and tximport") for an explanation of how to derive a count matrix from `Salmon`'s transcript-level quantification using `tximport`
[^2]: From page 250 of the [`limma` reference manual](https://bioconductor.org/packages/release/bioc/manuals/limma/man/limma.pdf)
[^3]: Note that in statistics, (numeric) data is often log-scaled when the data can take on large values